modules/doc/content/newsletter/2019_06.md

```diff
@@ -1,5 +1,92 @@
 # MOOSE News (June 2019)
 
+## Revamped Mortar FEM
+
+`MortarConstraint` has been revamped. Mortar now relies on an `AutomaticMortarGeneration` object
+that, as its name suggests, automatically generates a mortar segment mesh for performing
+integrations. For static mesh problems the mortar segment mesh generation is done once upon initial
+setup. For dynamic problems with a displaced mesh, mortar mesh generation is done every residual
+evaluation. We have verified that the new mortar system satisfies *a prior* error estimates, e.g.
+in the L2 norm we observe order 3 and order 2.5 convergence for the primal and lagrange multiplier
+variables respectively for a P2P1 discretization, as well as 2 and 1.5 for P1P1 and P1P0. Plots of
+these results as well as some technical discussion can be found on the mortar
+[github issue](https://github.com/idaholab/moose/issues/13080). Additionally, an excellent overview
+of the mortar implementation is given in
+[this technical report](https://www.osti.gov/biblio/1468630/). Examples of using the new mortar
+constraint can be found in the `$MOOSE_DIR/test/tests/mortar` directory. These examples demonstrate
+use of mortar with conforming and non-conforming meshes, as well as for periodic boundary
+conditions. Physics examples include solution continuity, thermal, and mechanical contact. In the
+latter case a finite difference Jacobian is used; however, a
+[follow-on PR](https://github.com/idaholab/moose/pull/13508) will demonstrate use of automatic
+differentiation and PJFNK for solving frictionless and frictional contact with the new mortar
+constraint system.
+
 ## PorousFlow: Reduction of Numerical Diffusion
 
 A new type of numerical stabilization has been added to PorousFlow to complement the existing "full upwind" and "no stabilization" choices.  The aim is to reduce numerical diffusion.  The code is based on the theoretical work of Kuzmin and Turek.  Users are strongly encouraged to experiment with this new feature so we can collectively learn the strengths and weaknesses in real simulations.  Only [minor changes to input files](porous_flow/kt.md) are needed to activate the new stabilization.  The numerical stabilization [lead page](porous_flow/stabilization.md) lists pros and cons of various stabilization schemes and provides links to pages that explain and quantify numerical diffusion.
+
+## Stochastic Tools "Batch" Mode
+
+The [Stochastic Tools module](stochastic_tools/index.md) now includes the ability to run
+full and transient solves using "batch" mode, which is useful when a MultiApp contains
+a large number of sub-applications. In batch mode the MultiApp creates a single sub-application
+and re-uses it to perform the multiple simulations, allowing for reduced memory usage when
+compared to the default behavior. For more information please refer to [batch_mode.md].
+
+## PETSc Upgrade
+
+PETSc is upgraded to 3.10.5 for moose users, and the new version provides a better performance.
```

Comment:
@fdkong Is the new version of PETSc available in the downloadable package? If so, this should just point people to download a new package. If not, then this section should be removed because building PETSc manually is not something many people do, so this will just be confusing.